packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
 import matplotlib.pyplot as plt
 from matplotlib import path
 
+from scipy.interpolate import RegularGridInterpolator
+
 import pydicom
 import pydicom.uid
 
-from .structure import pull_structure
-from .coords import coords_from_xyz_axes, xyz_axes_from_dataset
+from ..rtplan import get_surface_entry_point
 
+from .structure import pull_structure
+from .coords import xyz_axes_from_dataset
 
 # pylint: disable=C0103
 
 
 def zyx_and_dose_from_dataset(dataset):
     x, y, z = xyz_axes_from_dataset(dataset)
     coords = (z, y, x)
-    dose = dose_from_dataset(dataset, reshape=False)
+    dose = dose_from_dataset(dataset)
 
     return coords, dose
 
 
-def dose_from_dataset(ds, set_transfer_syntax_uid=True, reshape=True):
+def dose_from_dataset(ds, set_transfer_syntax_uid=True, reshape=False):
     r"""Extract the dose grid from a DICOM RT Dose file.
-
-    .. deprecated:: 0.5.0
-            `dose_from_dataset` will be removed in a future version of
-            PyMedPhys in favour of a new & improved API.
     """
 
     if set_transfer_syntax_uid:
         ds.file_meta.TransferSyntaxUID = pydicom.uid.ImplicitVRLittleEndian
 
     if reshape:
-        warnings.warn((
-            '`dose_from_dataset` currently reshapes the dose grid. In a '
-            'future version this will no longer occur. To begin using this '
-            'function without the reshape pass the parameter `reshape=False` '
-            'when calling `dose_from_dataset`.'), UserWarning)
-        pixels = np.transpose(
-            ds.pixel_array, (1, 2, 0))
-    else:
-        pixels = ds.pixel_array
+        warnings.warn(
+            ('The `reshape` parameter no longer does anything. Please remove '
+             'this parameter. In a future version this parameter will no '
+             'longer be accepted.'), UserWarning)
```

Comment:
Setting reshape to true now does nothing and warns that the parameter will be removed in the future.

Reply:
Why not remove now?

Reply:
Yeah, maybe it is worth removing it. That way, code that uses it will break instead of potentially silently continuing.